gp/genome.py

```diff
@@ -434,4 +454,35 @@ def clone(self) -> "Genome":
             list(self._primitives),
         )
         new.dna = self._dna.copy()
+
+        # Lamarckian strategy: parameter values are passed on to
+        # offspring
+        new.parameter_names_to_values = self.parameter_names_to_values.copy()
+
         return new
+
+    def update_parameters_from_torch_class(self, torch_cls: torch.nn.Module) -> None:
+        """Update values stored in Parameter nodes of graph from parameters of
+        a given Torch instance.  Can be used to import new values from
+        a Torch class after they have been altered, e.g., by local
+        search.
+
+        Parameters
+        ----------
+        torch_cls : torch.nn.module
+            Instance of a torch class.
+
+        Returns
+        -------
+        None
+
+        """
+        for name, value in torch_cls.named_parameters():
+            name = f"<{name[1:]}>"
+            if name in self.parameter_names_to_values:
+                self.parameter_names_to_values[name] = value.item()
+                assert not np.isnan(self.parameter_names_to_values[name])
+
+                # since a parameter of this individual has changed its
+                # fitness needs to be reevaluated
+                self.fitness = None
```

Comment:
this should be done in the individual not the genome. scary that no test catches this, will investigate tomorrow

Reply:
What's the outcome here?

Reply:
this was indeed wrong, fixed it now and added a test